yasa/hypno.py

```diff
@@ -48,6 +24,826 @@
+
+    .. versionadded:: 0.7.0
+
+    Parameters
+    ----------
+    values : array_like
+        A vector of stage values, represented as strings. See some examples below:
+
+        * 2-stages hypnogram (Wake/Sleep): ["W", "S", "S", "W", "S"]
+        * 3-stages hypnogram (Wake/NREM/REM): pd.Series(["W", "NREM", "NREM", "REM", "REM"])
+        * 4-stages hypnogram (Wake/Light/Deep/REM): np.array([["Wake", "Light", "Deep", "Deep"]])
+        * 5-stages hypnogram (default): ["N1", "N1", "N2", "N3", "N2", "REM", "W"]
+
+        Note that artefacts ("Art") and unscored ("Uns") epochs are always allowed regardless of the
+        number of stages in the hypnogram.
+    n_stages : int
+        Whether ``values`` comes from a 2, 3, 4 or 5-stages hypnogram. Default is 5, i.e. the
+        following stages are accepted: N1, N2, N3, REM, WAKE.
+    freq : str
+        A pandas frequency string indicating the frequency resolution of the hypnogram. Default is
+        "30s" meaning that each value in the hypnogram represents a 30-seconds epoch.
+        Examples: "1min", "10s", "15min".
+    start : str or datetime
+        An optional string indicating the starting date/time of the hypnogram. If ``start`` is
+        specified and valid, the index of the hypnogram will be a :py:class:`pandas.DatetimeIndex`.
+        e.g. "2022-12-15 22:30:00"
+    scorer : str
+        An optional striong indicating the scorer name. If specified, this will be set as the name
+        of the :py:class:`pandas.Series`, otherwise will be set to "Stage".
+
+    Examples
+    --------
+    Create a 2-stage hypnogram
+
+    >>> from yasa import Hypnogram
+    >>> values = ["W", "W", "W", "S", "S", "S", "S", "S", "W", "S", "S", "S"]
+    >>> hyp = Hypnogram(values, n_stages=2)
+    >>> hyp.hypno
+    0      WAKE
+    1      WAKE
+    2      WAKE
+    3     SLEEP
+    4     SLEEP
+    5     SLEEP
+    6     SLEEP
+    7     SLEEP
+    8      WAKE
+    9     SLEEP
+    10    SLEEP
+    11    SLEEP
+    Name: Stage, dtype: object
+
+    >>> hyp.n_epochs
+    12
+
+    >>> hyp.mapping
+    {'WAKE': 0, 'SLEEP': 1, 'ART': -1, 'UNS': -2}
+
+    >>> hyp.as_int()
+    0     0
+    1     0
+    2     0
+    3     1
+    4     1
+    5     1
+    6     1
+    7     1
+    8     0
+    9     1
+    10    1
+    11    1
+    Name: Stage, dtype: int64
+
+    >>> hyp.sleep_statistics()
+    {'TIB': 6.0,
+     'SPT': 4.5,
+     'WASO': 0.5,
+     'TST': 4.0,
+     'SE': 66.6667,
+     'SME': 88.8889,
+     'SFI': 7.5,
+     'SOL': 1.5,
+     'SOL_5min': nan,
+     'WAKE': 2.0}
+
+    >>> counts, probs = hyp.transition_matrix()
+    >>> counts
+    To Stage    WAKE  SLEEP
+    From Stage
+    WAKE           2      2
+    SLEEP          1      6
+
+    Simulate a 5-stages hypnogram
+
+    >>> from yasa import simulate_hypno
+    >>> values = simulate_hypno(tib=500, n_stages=5, seed=42)
+    >>> values = pd.Series(values).map({0: "W", 1: "N1", 2: "N2", 3: "N3", 4: "REM"}).to_numpy()
```

Comment:
Since we are passing strings into this class, we should probably switch it up so `yasa.simulate_hypno` returns strings (to avoid this step in many places).

Reply:
I was hoping you'd say that :) We can implement this in a future PR though, no rush